### PR TITLE
Ubuntu/mantic 23.4.x: Retain exit code in cloud-init status for recoverable errors

### PR DESCRIPTION
**Pull request by aciba90**
## do not squash

- [x] `quilt push -a && tox -e py3 && quilt pop -a`

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,11 @@
+cloud-init (23.4.1-0ubuntu1~23.10.1ubuntu1) UNRELEASED; urgency=medium
```
Comment (TheRealFalcon) [inline debian/changelog:1]: This should be `23.10.2` rather than `1ubuntu1`. File a bug against new_upstream_snapshot.py if we need one.

**Review by TheRealFalcon — APPROVED** [review]
Just need to update the version number, but otherwise LGTM.

Lets wait for standup before merging though.

File: debian/changelog (hunk 1 above)
Comment (blackboxsw) [reply debian/changelog:1]: Also please bump `UNRELEASED` to `mantic` so we can release this

**Review by blackboxsw — COMMENTED** [review]
Also right after we upload this let's get PRs against the base branches `upstream/ubuntu/(mantic|lunar|jammy|focal)` the sync both d/changelog and the new quilt patch and series updates into the base branches so we don't miss it on next  SRU

**Pull request merged**